DIPs/DIP1xxx-sw.md

```diff
@@ -0,0 +1,954 @@
+                        <identifierOrTemplateInstance>
+                            <identifier>F</identifier>
+                        </identifierOrTemplateInstance>
+                        <structInitializer>
+                          <structMemberInitializers>
+                            <structMemberInitializer>
+                              <identifier>a</identifier>
+                              <nonVoidInitializer>
+                                <unaryExpression>
+                                  <primaryExpression>
+                                    <intLiteral>1</intLiteral>
+                                  </primaryExpression>
+                                </unaryExpression>
+                              </nonVoidInitializer>
+                            </structMemberInitializer>
+                          </structMemberInitializers>
+                        </structInitializer>
+                    </structLiteral>
+
+                  </unaryExpression>
+                </argumentList>
+              </arguments>
+            </functionCallExpression>
+          </unaryExpression>
+        </expression>
+      </expressionStatement>
+    </statementNoCaseNoDefault>
+  </statement>
+</declarationOrStatement>
+```
+
+#### Bonus 1: Function calls with the struct argument as sole parameter
+
+Let's imagine we define a function `valid`:
+
+```d
+bool valid(S s)
+{
+    return s.c > 5;
+}
+```
+
+With the current options (e.g. [Option 3][option3]) calling `valid` would be written like this:
+
+```d
+valid(S(c: 10));
+```
+
+Thus, a potential addition would be, if
+
+- there's only one argument and
+- it's a struct
+- it doesn't have a user-defined constructor
+
+yielding syntax like this:
+
+```d
+valid(c: 10);
+```
+
+However, exploring this syntax goes over the scope of this DIP
+as this syntax might be used for named arguments in the future
+and if this syntax change turns out to be beneficial it could always extended in a further DIP.
+
+#### Bonus 2: Allowing in-place struct assignment
+
+This DIP does _not_ try to introduce the following syntax as it might be too ambiguous
+for parsers:
+
+    s = {c: 10};
+
+However, the proposed syntax would at least allow the following:
+
+    s = S({c: 10});
+
+Justifying whether it's worth to go one step more, is a task for another DIP.
+
+#### Other approaches
+
+##### Emulating named arguments with fat arrow templates
+
+Emulation of named arguments with fat arrows templates
+has been recently discovered by [Adam D. Ruppe](http://forum.dlang.org/post/awjuoemsnmxbfgzhgkgx@forum.dlang.org)
+with an implementation being in [ae.utils.meta.args](https://github.com/CyberShadow/ae/blob/master/utils/meta/args.d).
+For example, a function could be called like this:
+
+```d
+static int fun(int a=1, int b=2, int c=3, int d=4, int e=5)
+{
+	return a+b+c+d+e;
+}
+
+assert(args!(fun) == 15);
+assert(args!(fun, b=>3) == 16);
+```
+
+While this is very nice trick, it has a few downsides:
```

Comment:
I think a major reason is that it relies on function parameter names not changing, i.e. they become part of the library API. Named arguments need to be distinguished from normal parameters, which can be renamed - this DIP solves this problem.